components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -243,13 +244,19 @@ public static String getLoginPageURL(String clientId, String sessionDataKey,
             String selfPath = "/oauth2/authorize";
             AuthenticationRequest authenticationRequest = new AuthenticationRequest();
 
+            int tenantId = OAuth2Util.getClientTenatId();
+
             //Build the authentication request context.
             authenticationRequest.setCommonAuthCallerPath(selfPath);
             authenticationRequest.setForceAuth(forceAuthenticate);
             authenticationRequest.setPassiveAuth(checkAuthentication);
             authenticationRequest.setRelyingParty(clientId);
-            authenticationRequest.addRequestQueryParam(FrameworkConstants.RequestParams.TENANT_ID,
-                    new String[]{String.valueOf(OAuth2Util.getClientTenatId())});
+            try {
+                authenticationRequest.setTenantDomain(OAuth2Util.getTenantDomain(tenantId));
+            } catch (IdentityOAuth2Exception e) {
+                log.error("Error while getting tenant domain from tenant id", e);
+                throw new UnsupportedEncodingException("Error while getting tenant domain from tenant id");
```

Comment:
Did the required changes so that it throws IdentityOAuth2Exception itself. https://github.com/wso2/carbon-identity/pull/800/files#diff-36be4c39a0bee5ab80320f124bbce03bR232